src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_LENGTH = 10;
+    private T[] list;
+    private int size;
+
+    public ArrayList() {
+        list = (T[]) new Object[BASIC_LENGTH];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            addSpase();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Incorrect index of list");
+        } else {
+            if (index == size) {
+                /* If index of element fall into the end of array */
+                add(value);
+            } else {
+                if (list.length == size) {
+                    /* If array is overflowing, we need increase the array */
+                    addSpase();
+                    addByIndex(value, index);
+                    return;
+                }
+                addByIndex(value, index);
```

Comment:
` addByIndex(value, index);` is duplicated. Do we really need return in line 36? And try to simplify if else constructions 
`else { if ()` -> 'else if ()' when it is possible